src/device/usbd.c

```diff
@@ -633,7 +633,7 @@ static void mark_interface_endpoint(uint8_t ep2drv[8][2], uint8_t const* p_desc,
       ep2drv[tu_edpt_number(ep_addr)][tu_edpt_dir(ep_addr)] = driver_id;
     }
 
-    len   += tu_desc_len(p_desc);
+    len   = (uint16_t)(len + tu_desc_len(p_desc));

```

Comment:
how about `len += (uint16_t) tu_desc_len(p_desc)

Reply:
My compiler complains on pedantic mode... The issue is that the (+) operation widens the result to integer (32-bit), but it's being stored to a 16-bit value, which means that it could be an overflow.

I'll revert this.

Reply:
compiler nowadays is even smarter than us :). I am fine with either way.